include/envoy/stats/stats_matcher.h

```diff
@@ -19,6 +19,16 @@ class StatsMatcher {
    * @return bool true if that stat should not be instantiated.
    */
   virtual bool rejects(const std::string& name) const PURE;
+
+  /**
+   * @return bool whether StatsMatcher trivially accepts all stats.
```

Comment:
I think "trivially" could use a bit more explanation, since as your unit tests show it's possible to acceptAll with acceptsAll returning false

If it's not easy to explain at a pure api-and-not-impl level maybe making it more clear that if this returns true the matcher acceptsAll but if the matcher acceptsAll this may or may not return true